students/john_hunter/lesson03/mailroom.py

```diff
@@ -0,0 +1,219 @@
+    maxLenOfName=0
+    
+    for i in range(1,len(donors)):                     
+        if (len(donors[i][0])>maxLenOfName):    
+            maxLenOfName = len(donors[i][0])
+
+    while True:
+        inputName= input('Enter full name of donor or \'list\' to request list of donors: ')  
+        for j in range(len(donors)):
+            names.append(donors[j][0])  
+        
+        if inputName == 'list':        
+            for k in range(len(donors)):
+                name = donors[k][0]
+                print('{:{align}{width}}'.format(name, align='<', width=maxLenOfName))
+            else: 
+                continue
+        
+        if inputName == 'exit':
+            quit_it()
+        
+        if inputName in names:
+            listOrNot = 'y'
+            break
+        
+        else :
+            listOrNot = 'n'
+        if listOrNot == 'n':
+            add_donor(inputName)
+            listOrNot = 'n'
+            moreDonations = 'n'
+            break
+    masDonations = 'Would you like to add more donations for ' + inputName + '? y/n: ' 
+    if listOrNot == 'y':
+        moreDonations = input(masDonations)    
+    if moreDonations == 'y':
+        totals = add_donations(inputName)
+    else:
+        totals = 0
+        for z in range(len(donors)):
+            if inputName == donors[z][0]:
+                for c in range(len(donors[z])-1):
+                    totals = totals + int(donors[z][c+1])
+                break
+
+    
+    emailText1 = "Dear " + inputName
+    emailText2 = "Thank you for your generous donation(s) of $" + str(totals) + '.'
+    emailText3 = "Sincerely," + '\n' + 'John Hunter'
+    print(emailText1 + '\n'*2 + emailText2 + '\n'*2 + emailText3 )
+
+def add_donor(name):
+    """Adds a donor to the donor dictionary and allows the user to add donation values
+    
+    Returns: 'None'
+    """
+    choice = 'y'
+    donations = list()
+    print('The name submitted is not on the list of donors.')
+    print('The donor name will be added, please add donation values: ')
+    while choice == 'y':
+        donation = input('Enter a donation amount, enter \'0\' to stop adding donation values: ')
+        if donation == '0':
+            choice = False
+        else:
+            donations.append(donation)
+    donations.insert(0,name)
+    newDonor = donations
+    donors.append(newDonor)
+    print('The new donor has been added:')
+    print(newDonor)
+    print('The Thank You email for the new donor is:')
+    return None 
+
+def add_donations(name):
+    donations = list()
+    donation = int()
+    while True:
+        donation = int(input('Enter a donation amount, enter \'0\' to stop adding donation values: '))
+        if donation == 0:
+            break
+        else:
+            int(donation)
+            donations.insert(-1,donation)
+    for i in range(len(donors)):
+        if name == donors[i][0]:
+            donors[i] = donors[i] + donations
+            break
```

Comment:
there is no need to use range() to loop over your list:
```
for donor in donors:
    if name == donor[0]:
        donor.append(donation)
        break
else:
    donors.append([name, donation])
```

this is much simpler ^^ and covers a lot of logic across `add_donations` and `add_donor` functions.